src/Forms/DateField.php

```diff
@@ -58,175 +33,320 @@
  *
  *   $f = new DateField('MyDate');
  *   $f->setLocale('de_DE');
- *   $f->setConfig('dmyfields', true);
+ *   $f->setSeparateDMYFields(true);
  *
  * # Validation
  *
  * Caution: JavaScript validation is only supported for the 'en_NZ' locale at the moment,
  * it will be disabled automatically for all other locales.
+ *
+ * # Formats
+ *
+ * All format strings should follow the CLDR standard as per
+ * http://userguide.icu-project.org/formatparse/datetime. These will be converted
+ * automatically to jquery UI format.
+ *
+ * The value of this field in PHP will be ISO 8601 standard (e.g. 2004-02-12), and
+ * stores this as a timestamp internally.
+ *
+ * Note: Do NOT use php date format strings. Date format strings follow the date
+ * field symbol table as below.
+ *
+ * @see http://userguide.icu-project.org/formatparse/datetime
+ * @see http://api.jqueryui.com/datepicker/#utility-formatDate
  */
 class DateField extends TextField
 {
-
     protected $schemaDataType = FormField::SCHEMA_DATA_TYPE_DATE;
 
     /**
-     * @config
-     * @var array
+     * Override locale. If empty will default to current locale
+     *
+     * @var string
+     */
+    protected $locale = null;
+
+    /**
+     * Override date format. If empty will default to that used by the current locale.
+     *
+     * @var null
      */
-    private static $default_config = array(
-        'showcalendar' => false,
-        'jslocale' => null,
-        'dmyfields' => false,
-        'dmyseparator' => '&nbsp;<span class="separator">/</span>&nbsp;',
-        'dmyplaceholders' => true,
-        'dateformat' => null,
-        'datavalueformat' => 'yyyy-MM-dd',
-        'min' => null,
-        'max' => null,
-    );
+    protected $dateFormat = null;
 
     /**
-     * @var array
+     * Set if js calendar should popup
+     *
+     * @var bool
      */
-    protected $config;
+    protected $showCalendar = false;
```

Comment:
One of the reasons I've avoided elevating these settings to class properties is that I didn't want to lock in the API to that extent - it's a bit like using `FormField->setAttributes()` to influence client-side react component presentation.

So rather than increasing the API surface of `DateField`, what do you think about separating out `DateFieldSeparated extends DateField` and `DateFieldJQuery extends DateField`? Then we can set the `DateField` injector service name to resolve to`DateFieldJQuery` by default, in order to get correct behaviour for built-in features like `DBDate->scaffoldFormField()`